app/src/main/java/dev/arkbuilders/arkmemo/di/RepositoryModule.kt

```diff
@@ -27,12 +29,15 @@ abstract class RepositoryModule {
 
     @Binds
     abstract fun bindVoiceNotesRepo(impl: VoiceNotesRepo): NotesRepo<VoiceNote>
+}
 
-    companion object {
-        @Provides
-        fun provideNotesRepoHelper(
-            memoPreferences: MemoPreferences,
-            propertiesStorageRepo: PropertiesStorageRepo,
-        ) = NotesRepoHelper(memoPreferences, propertiesStorageRepo)
-    }
+@InstallIn(SingletonComponent::class)
+@Module
+object RepoHelperModule {
+    @Singleton
+    @Provides
+    fun provideNotesRepoHelper(
+        propertiesStorageRepo: PropertiesStorageRepo,
+        @Named(IO_DISPATCHER) coroutineDispatcher: CoroutineDispatcher,
```

Comment:
**suggestion:** Let's not use a coroutine with IO to access shared preferences as it potentially causes leak. Access it normally is good enough.

https://stackoverflow.com/questions/58536773/is-it-ok-to-use-sharedprefrence-with-coroutine-kotlin

Reply:
@hieuwu This change provides a `CoroutineDispatcher` when initializing a `NotesRepoHelper`, and removes unnecessary dependency of `MemoPreferences`.
Could you elaborate more on how it's related to your comment please?

Reply:
Oh I misread the usage. All good now 👍